Add cpu morpher

zbai-sc opened the pull request:
- add cpu morpher which supports more than 4 active targets
- add common interface for morphers
- change Animator to allow swapping out the stock MorphHelper for the
new CPUMorpher
- change Animator to allow morphers to be managed externally

elisemorysc reviewed (commented):
Most of the code here is copy and paste from filament. Can't we reuse that code?

zbai-sc commented:
> Most of the code here is copy and paste from filament. Can't we reuse that code?

For ReusableAnimator, the problem is that the way the structured their Animator class make it impossible to extended their class as their hide most of the actual logic inside an internal class. On the other hand, changing Animator would result in a much bigger change in the code base as it was being used in a lot of different places. It's best we keep our code separate from their stuff for now.

For CPUMorpher, we actually changed the logic almost completely.

zbai-sc commented:
The last [commit](https://github.com/elisemorysc/filament/pull/7/commits/21e2436653b724d15f79a580589901877f5593d1) turns out not to be the issue (I found an issue on hermes side regarding destruction sequences). But nice to keep it in there for precaution.

elisemorysc commented:
I think you made a mistake in the PR request? Wrong target branch?

zbai-sc commented:
> I think you made a mistake in the PR request? Wrong target branch?

This change depends on the animator change as it involves some changes to the Animator. I'm targeting the Animator PR so it only show the changes regarding the morpher. Will retarget huner-main once we merge the other PR.

elisemorysc reviewed (approved):
LGTM

elisemorysc commented on filament/backend/src/opengl/OpenGLDriver.cpp:
```diff
@@ -1275,7 +1275,10 @@ void OpenGLDriver::createTimerQueryR(Handle<HwTimerQuery> tqh, int) {
 void OpenGLDriver::destroyVertexBuffer(Handle<HwVertexBuffer> vbh) {
     DEBUG_MARKER()
-    // Do nothing, because a VertexBuffer is just a collection of BufferObject handles.
+    if (vbh) {
```
This is part of hunter-main. Do we need this in the PR?

zbai-sc replied:
Removed. Thanks for updating hunter-main!

zbai-sc committed:
Enhance morpher

- add cpu morpher which supports more than 4 active targets
- add common interface for morphers
- change Animator to allow swapping out the stock MorphHelper for the
new CPUMorpher
- change Animator to allow morphers to be managed externally
